app/docs/md/patterns/event-listeners.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Event Listeners
+---
+
+An event is a signal that something has happened on your page. The browser notifies you so you can react to them.
+
+## What are events?
+
+From [MDN](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#what_is_an_event)
+
+> Events are things that happen in the system you are programming — the system produces (or "fires") a signal of some kind when an event occurs, and provides a mechanism by which an action can be automatically taken (that is, some code running) when the event occurs. Events are fired inside the browser window, and tend to be attached to a specific item that resides in it. This might be a single element, a set of elements, the HTML document loaded in the current tab, or the entire browser window. There are many different types of events that can occur.
+
+For example:
+
+- A user submits a form
+- A user clicks on a button
+- A web page finishes loading
+- etc.
+
+**[Learn more about events](https://developer.mozilla.org/en-US/docs/Web/Events)**
+
+## Adding a click event listener
+
+Suppose we have a simple web component called `custom-button` that we want to add a click event listener to.
```

Comment:
Thinking this should maybe just be a vanilla WC rather than extending `CustomElement`, since we're not really gaining anything from it (except for style scoping, which we likely don't need to show here). Since events are so basic I feel like adding an abstraction off the bat may confuse some folks 🤔

Reply:
I could go either way on this one. The user could be turned off by the amount of boilerplate needed if we show the stock web component.

Reply:
Hmm, yeah since this is a client side component I guess so — an SSR element would be simpler but then we'd have to explain that whole divide. Fair!

Reply:
I worry about doing a vanilla WC here as we'd lose the ability to SSR it and it wouldn't be consistent with the rest of the site.

Reply:
All good! Will close this one.